qa/src/test/java/org/eclipse/kapua/service/datastore/steps/DataStoreServiceSteps.java

```diff
@@ -479,6 +551,42 @@ public void queryForAccountChannelInfoByDateRange(String start, String end, Stri
         stepData.put(lstKey, tmpList);
     }
 
+    @When("^I query for the current account channels with the filter \"(.+)\" and store them as \"(.+)\"$")
+    public void queryForAccountChannelsWithFiltering(String filter, String lstKey) throws KapuaException {
+
+        Account account = (Account) stepData.get("LastAccount");
+        DatastoreChannel datastoreChannel = new DatastoreChannel(filter);
+        ChannelInfoQuery tmpQuery = DatastoreQueryFactory.createBaseChannelInfoQuery(account.getId(), 100);
+        ChannelMatchPredicate channelMatchPredicate = new ChannelMatchPredicateImpl(datastoreChannel.getChannelCleaned());
+        AndPredicate andPredicate = new AndPredicateImpl();
+        andPredicate.getPredicates().add(channelMatchPredicate);
+        tmpQuery.setPredicate(andPredicate);
+
+        ChannelInfoListResult tmpList = channelInfoRegistryService.query(tmpQuery);
+        stepData.put(lstKey, tmpList);
+    }
+
+    @Then("^The channel list \"(.+)\" contains the following topics$")
+    public void checkThattheChannelInfoContainsTheSpecifiedTopics(String lstKey, List<TestTopic> topics) {
+
+        ChannelInfoListResult tmpList = (ChannelInfoListResult) stepData.get(lstKey);
+        Set<String> infoTopics = new HashSet<>();
+        Set<String> listTopics = new HashSet<>();
+
+        assertEquals("Wrong number of topics found!", tmpList.getSize(), topics.size());
+
+        for (ChannelInfo tmpInfo : tmpList.getItems()) {
+            infoTopics.add(tmpInfo.getName());
+        }
+        for (TestTopic tmpTop : topics) {
+            listTopics.add(tmpTop.getTopic());
+        }
+
+        for (String tmpTopic : listTopics) {
```

Comment:
I think we can remove the lines 581/583 since we can use the topics list into this loop